Assets/Scripts/FinalBoss/DaggerPower.cs

```diff
@@ -2,17 +2,23 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class DaggerPower : MonoBehaviour
+public class DaggerPower : Gun
 {
-    // Start is called before the first frame update
-    void Start()
+    [SerializeField] private Transform daggerProj;
+    override protected void Update()
     {
-        
-    }
+        base.Update();
 
-    // Update is called once per frame
-    void Update()
-    {
-        
+        if (Input.GetMouseButton(0) && Time.time - lastFireTime > fireRate)
+        {
+            lastFireTime = Time.time;
+            FireProjectile(Constants.PickUpWater, daggerProj);
+        }
+
+        if (aim.rightStickInput.magnitude > 0.5 && Time.time - lastFireTime > fireRate)
+        {
+            lastFireTime = Time.time;
+            FireProjectile(Constants.PickUpWater, daggerProj);
+        }
```

Comment:
this looks like the same code for all the types, just a different variables passed into the FireProjectile function. Could we create a base function inside Gun and just have local variable that is set for PickupType and projectileObj then use those in the update rather than duplicate this code in 3 files?